tests/DemosTest.php

```diff
@@ -502,7 +503,12 @@ public function provideDemoAssertJsonResponsePostCases(): iterable
      */
     public function testDemoAssertJsonResponsePost(string $path, array $postData): void
     {
-        $response = $this->getResponseFromRequest($path, ['form_params' => $postData]);
+        $response = $this->getResponseFromRequest($path, [
+            'headers' => [
+                'Content-Type' => 'application/x-www-form-urlencoded',
```

Comment:
From ServerRequestCreator

_POST must be loaded only if `Content-Type` equals to : `application/x-www-form-urlencoded` OR `multipart/form-data`

In Browser, this is done internally.
In Guzzle probably leave to the user all options, even bad formatted requests.
I think We can add Content-Type if POST refactoring it in our code: https://github.com/atk4/ui/blob/d64706e9f8cf3eab4b1b64cadea8eb1b6881774e/tests/DemosTest.php#L199

```php
    public function fromGlobals(): ServerRequestInterface
    {
        $server = $_SERVER;
        if (false === isset($server['REQUEST_METHOD'])) {
            $server['REQUEST_METHOD'] = 'GET';
        }

        $headers = \function_exists('getallheaders') ? getallheaders() : static::getHeadersFromServer($_SERVER);

        $post = null;
        if ('POST' === $this->getMethodFromEnv($server)) {
            foreach ($headers as $headerName => $headerValue) {
                if (true === \is_int($headerName) || 'content-type' !== \strtolower($headerName)) {
                    continue;
                }
                if (\in_array(
                    \strtolower(\trim(\explode(';', $headerValue, 2)[0])),
                    ['application/x-www-form-urlencoded', 'multipart/form-data']
                )) {
                    $post = $_POST;

                    break;
                }
            }
        }

        return $this->fromArrays($server, $headers, $_COOKIE, $_GET, $post, $_FILES, \fopen('php://input', 'r') ?: null);
    }
```

Reply:
Yes, the reason is https://github.com/Nyholm/psr7-server/blob/1.0.2/src/ServerRequestCreator.php#L62 - next time, please post a link instead of copy pasting the code here... Anyway, see one of my latest commit, when all request headers are reconstructed into $_SERVER, no additional code is needed